pkg/awsutils/awsutils_test.go

```diff
@@ -605,29 +639,52 @@ func TestAllocIPAddresses(t *testing.T) {
 	defer ctrl.Finish()
 
 	// when required IP numbers(5) is below ENI's limit(30)
+	eniID := "eni-id"
 	input := &ec2.AssignPrivateIpAddressesInput{
-		NetworkInterfaceId:             aws.String("eni-id"),
+		NetworkInterfaceId:             aws.String(eniID),
 		SecondaryPrivateIpAddressCount: aws.Int64(5),
 	}
 	mockEC2.EXPECT().AssignPrivateIpAddressesWithContext(gomock.Any(), input, gomock.Any()).Return(nil, nil)
 
 	ins := &EC2InstanceMetadataCache{ec2SVC: mockEC2, instanceType: "c5n.18xlarge"}
-	err := ins.AllocIPAddresses("eni-id", 5)
+	err := ins.AllocIPAddresses(eniID, 5)
 	assert.NoError(t, err)
 
 	// when required IP numbers(50) is higher than ENI's limit(49)
 	input = &ec2.AssignPrivateIpAddressesInput{
-		NetworkInterfaceId:             aws.String("eni-id"),
+		NetworkInterfaceId:             aws.String(eniID),
 		SecondaryPrivateIpAddressCount: aws.Int64(49),
 	}
-	mockEC2.EXPECT().AssignPrivateIpAddressesWithContext(gomock.Any(), input, gomock.Any()).Return(nil, nil)
+	addresses := make([]*ec2.AssignedPrivateIpAddress, 49)
+	output := ec2.AssignPrivateIpAddressesOutput{
+		AssignedPrivateIpAddresses: addresses,
+		NetworkInterfaceId:         aws.String(eniID),
+	}
+	mockEC2.EXPECT().AssignPrivateIpAddressesWithContext(gomock.Any(), input, gomock.Any()).Return(&output, nil)
 
 	ins = &EC2InstanceMetadataCache{ec2SVC: mockEC2, instanceType: "c5n.18xlarge"}
-	err = ins.AllocIPAddresses("eni-id", 50)
+	err = ins.AllocIPAddresses(eniID, 50)
 	assert.NoError(t, err)
 
 	// Adding 0 should do nothing
-	err = ins.AllocIPAddresses("eni-id", 0)
+	err = ins.AllocIPAddresses(eniID, 0)
+	assert.NoError(t, err)
+}
+
+func TestAllocIPAddressesAlreadyFull(t *testing.T) {
+	ctrl, mockEC2 := setup(t)
+	defer ctrl.Finish()
+	// The required IP numbers(14) is the ENI's limit(14)
+	input := &ec2.AssignPrivateIpAddressesInput{
+		NetworkInterfaceId:             aws.String("eni-id"),
```

Comment:
nit: should we use the consistent name "eniID"?

Reply:
Thanks @haouc, addressed